pkg/crd/testdata/cronjob_types.go

```diff
@@ -162,9 +162,13 @@ type CronJobSpec struct {
 	StructWithSeveralFields NestedObject `json:"structWithSeveralFields"`
 
 	// A struct that can only be entirely replaced via a nested type.
-	// +structType=atomic
 	NestedStructWithSeveralFields NestedStructWithSeveralFields `json:"nestedStructWithSeveralFields"`
 
+	// A struct that can only be entirely replaced via a nested type and
+	// field markers.
+	// +structType=atomic
+	NestedStructWithSeveralFieldsDoubleMarked NestedStructWithSeveralFields `json:"nestedStructWithSeveralFieldsDoubleMarked"`
```

Comment:
It seems a workarround.  Could you please raise an issue or add in the description:

a) what you were doing when you face the error ( with an example code )
b) what do you expected to see ?
c) what was the behaviour faced instead? 

Also, if we move forward here I think we need a better name than `FieldsDoubleMarked`. DoubleMarket with what? (i.e. NestedStructWithSeveralFieldsMarkedWithAB)

Reply:
The context of this is https://github.com/kubernetes-sigs/controller-tools/issues/676, which is a duplicate of https://github.com/kubernetes-sigs/controller-tools/issues/435.

Reply:
@camilamacedo86 so this just adds an additional test case. `NestedStructWithSeveralFields` ensures that the resulting manifest carries the `atomic` marker from the type. `NestedStructWithSeveralFieldsDoubleMarked` ensures the same thing only that the marker is applied to the type, as well as the to the field of the struct.